### PR TITLE
feat(MedicationManager): prevent addition of duplicate medications

### LINKED ISSUE #168: [PE-D][Tester A] Added medication can be repeated
The application allows the user to add the same medication again into the list. This could cause confusion for the user. It would be great to check for same medication added so they cannot be added to the list again.

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: Haotian199/ped#2

### LINKED ISSUE #190: [PE-D][Tester F] adding the exact same medication could possibly add onto the existing input rather than listing it as a new one
-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: nikhil-2101/ped#4

### PR DESCRIPTION
**Pull request by kyuichyi**
# Checklist for feature implementation and modification
Closes #190, Closes #168 
- [ ] Added relevant milestone (under milestone tab on the right)
- Implementation
  - [ ] Sources (if any) credited under both relevant section(s) and Developer Guide (Acknowledgement)
  - [ ] JUnit Test(s)
- Documentation
  - [x] JavaDoc for classes and methods
  - Developer Guide
    - [ ] Design (Can add as a todo statement for later on)
    - [ ] Implementation (Can add as a todo statement for later on)
    - [ ] Relevant User Story present
    - [ ] Non-functional requirement(s) that the feature addresses (if any)
    - [ ] Glossary of the (technical) term(s) used (if any)
    - [ ] Specialised instructions for manual testing (if any)
  - User Guide
    - [ ] Description and usage of feature (if any)
  - Contributions (under `/docs/team/*.md`)
    - [ ] Code/Feature/Enhancement contribution
    - [ ] Developer Guide contribution
    - [ ] User Guide contribution
    - [ ] Misc and team-based admin tasks (if any)

# Checklist for non-implementation related PR
Closes #<Issue_Number>
- [ ] Added relevant milestone (under milestone tab on the right)
- Documentation (Contributions)
  - [ ] Developer Guide contributions (if any)
  - [ ] User Guide contributions (if any)
  - [ ] Review contributions (if any)
  - [ ] Miscellaneous and team-based admin tasks (if any)

# Additional Information

Nil

### CODE REVIEW

File: src/main/java/meditracker/medication/MedicationManager.java [hunk 1]
```diff
@@ -49,12 +50,28 @@ protected static void clearMedication() {
      * Adds a Medication to the list of Medication
      *
      * @param medication Medication to be added to the list
+     * @throws MediTrackerException When a duplicate medication is found
      */
-    public static void addMedication(Medication medication) {
+    public static void addMedication(Medication medication) throws MediTrackerException{
```
Comment (annoy-o-mus) [inline src/main/java/meditracker/medication/MedicationManager.java:55]: Please add a space before the opening `{`

File: src/main/java/meditracker/exception/MediTrackerException.java [hunk 2]
```diff
@@ -14,4 +14,8 @@ public class MediTrackerException extends Exception {
     public MediTrackerException(String errorContext) {
         super(errorContext);
     }
+
+    public MediTrackerException() {
```
Comment (annoy-o-mus) [inline src/main/java/meditracker/exception/MediTrackerException.java:18]: Is this empty constructor being used by the code?

File: src/main/java/meditracker/exception/DuplicateArgumentFoundException.java [hunk 3]
```diff
@@ -7,4 +7,7 @@ public class DuplicateArgumentFoundException extends Exception {
     public DuplicateArgumentFoundException(String errorContext) {
         super(errorContext);
     }
+
+    public DuplicateArgumentFoundException() {
```
Comment (annoy-o-mus) [inline src/main/java/meditracker/exception/DuplicateArgumentFoundException.java:11]: Is this empty constructor being used by the code?

**Review by annoy-o-mus — CHANGES_REQUESTED** [review]
Great job, just some minor clarafications needed.

File: src/main/java/meditracker/exception/DuplicateArgumentFoundException.java (hunk 3 above)
Comment (kyuichyi) [reply src/main/java/meditracker/exception/DuplicateArgumentFoundException.java]: Resolved in https://github.com/AY2324S2-CS2113T-T09-1/tp/pull/218/commits/f600cce87c24fdfec92f3bd1c50836666b14f4fd

File: src/main/java/meditracker/exception/MediTrackerException.java (hunk 2 above)
Comment (kyuichyi) [reply src/main/java/meditracker/exception/MediTrackerException.java]: Resolved in https://github.com/AY2324S2-CS2113T-T09-1/tp/pull/218/commits/f600cce87c24fdfec92f3bd1c50836666b14f4fd

File: src/main/java/meditracker/medication/MedicationManager.java (hunk 1 above)
Comment (kyuichyi) [reply src/main/java/meditracker/medication/MedicationManager.java]: Resolved in https://github.com/AY2324S2-CS2113T-T09-1/tp/pull/218/commits/6bad8f6bb17510a2145b6373685aa1dfcef58c3a

**Review by annoy-o-mus — APPROVED** [review]

**Pull request merged**